### PR TITLE
ARROW-1875: [Java] Write 64-bit ints as strings in integration test JSON files

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-1875](https://issues.apache.org/jira/browse/ARROW-1875).
This is Java side implementation.

### CODE REVIEW

**Comment by emkornfield** [comment]
It is surprising that this passes, I will need to check it out and run it locally to make sure something funny isn't happening. But then i think it can be merged.

File: java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java [hunk 1]
```diff
@@ -378,7 +379,7 @@ protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException
 
         for (int i = 0; i < count; i++) {
           parser.nextToken();
-          BigInteger value = parser.getBigIntegerValue();
+          BigInteger value = new BigInteger(parser.getValueAsString());
```
Comment (praveenbingo) [inline java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: will this not overflow doing a biginteger.longValue()? or change sign.

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: I guess not, when write UINT8 values with JsonFileReader, it uses getNoOverflow which is already BigInteger?

Comment (praveenbingo) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: no i mean the line below "buf.writeLong(value.longValue())"

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: Seems not, I will take a close watch.

Comment (praveenbingo) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: Ok went through UINT8Vector and saw the method implementation, i think there should be no problems like you said too.(although a test case would be nice, nothing blocking though).

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java]: Sure, I could add some tests for UINT type vectors in a seperate issue :)

**Comment by tianchen92** [comment]
> It is surprising that this passes, I will need to check it out and run it locally to make sure something funny isn't happening. But then i think it can be merged.

Hi Micah, Anything found？

**Comment by emkornfield** [comment]
OK, the reason why this doesn't break other languages is the integration tests don't seem to actually use the write path (arrow->json), and the read path (json->arrow) is robust to either strings or numbers.  So I think this can be merged.  And we can have other languages make similar changes and then change the python script.

CC @wesm

**Comment by emkornfield** [comment]
+1 thank you.

**Pull request closed without merging**